kurtosis-devnet/book/src/basic_deployment.md

```diff
@@ -0,0 +1,138 @@
+# Basic Deployment
+
+The Kurtosis devnet provides several pre-configured devnet templates and convenient commands to deploy and interact with them.
+
+## Built-in Devnets
+
+The following devnet templates are available out of the box:
+
+1. **Simple Devnet** (`simple.yaml`)
+   - Basic single-chain setup
+   - Ideal for local development and testing
+   - Deploy with: `just simple-devnet`
+
+2. **Interop Devnet** (`interop.yaml`)
+   - Designed for interop testing
+   - Includes test suite for cross-chain interactions
+   - Deploy with: `just interop-devnet`
+   - Run tests with: `just interop-devnet-test`
+
+3. **Pectra Devnet** (`pectra.yaml`)
+   - Specialized configuration for Pectra testing
+   - Deploy with: `just pectra-devnet`
+
+## User-Defined Devnets (Experimental)
+
+> **Note**: User-defined devnets are an experimental feature and not actively supported at this time. Use at your own risk.
+
+The user devnet template (`user.yaml`) allows for customizable devnet configurations through a JSON input file. This feature is designed to simplify devnet creation for future devnet-as-a-service scenarios.
+
+### Deployment
+```bash
+just user-devnet <data-file>
+```
+
+### Example Configuration
+Here's an example of a user devnet configuration file:
+
+```json
+{
+    "interop": true,
+    "l2s": {
+        "2151908": {
+            "nodes": ["op-geth", "op-geth"]
+        },
+        "2151909": {
+            "nodes": ["op-reth"]
+        }
+    },
+    "overrides": {
+        "flags": {
+            "log_level": "--log.level=debug"
+        }
+    }
+}
+```
+
+This configuration:
+- Enables interop testing features
+- Defines two L2 chains:
+  - Chain `2151908` with two `op-geth` nodes
+  - Chain `2151909` with one `op-reth` node
+- Sets custom logging level for all nodes
+
+## Deployment Commands
+
+Arbitrary devnets can be deployed using the general `devnet` command with the following syntax:
+```bash
+just devnet <template-file> [data-file] [name]
+```
+
+Where:
+- `template-file`: The YAML template to use (e.g., `simple.yaml`)
+- `data-file`: Optional JSON file with configuration data
+- `name`: Optional custom name for the devnet (defaults to template name)
+
+For example:
+```bash
+# Deploy simple devnet with default name
+just devnet simple.yaml
+
+# Deploy user devnet with custom data and name
+just devnet user.yaml my-config.json my-custom-devnet
+```
+
+This can be convenient when experimenting with devnet defitions
+
+## Entering a Devnet Shell
+
+The devnet provides a powerful feature to "enter" a devnet environment, which sets up the necessary environment variables for interacting with the chains.
+
+### Basic Usage
+```bash
+just enter-devnet <devnet-name> [chain-name]
+```
```

Comment:
Is this a redundant CLI with shell integration of the devnet-sdk, or possibly it will expose kurtosis-specific options?

Reply:
that's entirely based on the devnet-sdk feature, just integrated with our build targets for convenience. It basically constructs the URL for the devnet based on our naming conventions for the enclave, and hands over to devnet-sdk right away.

Reply:
I see. Thank you.